String: rename `split` without separator to `words`

MakeNowJust opened the pull request:
See https://github.com/crystal-lang/crystal/issues/4644#issuecomment-312295584

Now, `String#split` without separator is deprecated (breaking change).

straight-shoota commented:
I think it could be argued to *add* `String#words()` besides `String.split()` with a handling leading and trailing whitespace differenty. But removing `String.split()` makes no sense to me.

need47 commented:
I think we just need a clear documentation on this, and that's enough.

MakeNowJust commented:
@straight-shoota We can use `s.split(/\s+/)` explicitly if need such a result. Of course, there is performance issue, however I don't know when such a behavior is needed.

MakeNowJust commented:
To fit `String#split` without separator semantics, Ruby decided to remove leading and trailing empty strings from other `split` result. However Crystal decided to keep them. At that time, `String#split` without separator is broken.

RX14 commented:
I believe `String#split` should behave exactly the same as `String#split(' ')` (except for splitting on all *unicode* whitespace (current is ascii-only)). Adding a `String#words` with the old implementation would be a good idea.

MakeNowJust commented:
We take care of `String#split`'s default separator. For example, in JavaScript `String.prototype.split` makes an array containing the entire string if separator is not given. You may believe `String#split`'s default separator is whitespace and it is the only answer without reason, but I am not believer. I think `String#split` should be given separator explicitly.

However, in fact, strangely, current Crystal behavior is almost same as Python's one (Python does not support empty string separator, but others are same). As @need47 said, we may have to write a clear documentation [like Python](https://docs.python.org/3/library/stdtypes.html#str.split). I agree with it if prefer simple solution.

bew commented:
For others not following everything, this PR has been closed because [this comment](https://github.com/crystal-lang/crystal/issues/4644#issuecomment-312438127).
